Neos.ContentRepository.Core/Classes/NodeType/NodeType.php

```diff
@@ -155,6 +155,29 @@ protected function buildFullConfiguration(): array
             $this->fullConfiguration['childNodes'] = $sorter->toArray();
         }
 
+        // migrate old property like references to references
+        $referencesConfiguration = $this->fullConfiguration['references'] ?? [];
+        foreach ($this->fullConfiguration['properties'] ?? [] as $propertyName => $propertyConfiguration) {
+            $propertyType = $propertyConfiguration['type'] ?? 'string';
+            switch ($propertyType) {
+                case 'reference':
+                    unset($propertyConfiguration['type']);
+                    $propertyConfiguration['constraints']['maxItems'] = 1;
+                    // @deprecated remove with 10
+                    // used to ensure that the FlowQuery property operation will return the node directly but not an array of nodes
+                    $propertyConfiguration['__legacyPropertyType'] = 'reference';
```

Comment:
that might need to be discussed?

Reasoning is that flowquery handled multiple references and single references different in the past.
With the new syntax there is always multiple ones and maybe a limit to 1 or any other value.

To make `q(node).property('myLegacyReferenceProperty')` return the node directly but `q(node).property('referenceWithLimit1')`and array of nodes, we need this distinction.

Reply:
For legacy we can also just look into `constraints.maxItems: 1`